currencies/rate_converter.go

```diff
@@ -30,55 +32,23 @@ func NewRateConverter(
 	fetchingInterval time.Duration,
 	staleRatesThreshold time.Duration,
 ) *RateConverter {
-	return NewRateConverterWithNotifier(
-		httpClient,
-		syncSourceURL,
-		fetchingInterval,
-		staleRatesThreshold,
-		nil, // no notifier channel specified, won't send any notifications
-	)
-}
-
-// NewRateConverterDefault returns a RateConverter with default values.
-// By default there will be no currencies conversions done.
-// `currencies.ConstantRate` will be used.
-func NewRateConverterDefault() *RateConverter {
-	return NewRateConverter(&http.Client{}, "", time.Duration(0), time.Duration(0))
-}
-
-// NewRateConverterWithNotifier returns a new RateConverter
-// it allow to pass an update chan in which the number of ticks will be passed after each tick
-// allowing clients to listen on updates
-// Do not use this method
-func NewRateConverterWithNotifier(
-	httpClient httpClient,
-	syncSourceURL string,
-	fetchingInterval time.Duration,
-	staleRatesThreshold time.Duration,
-	updateNotifier chan<- int,
-) *RateConverter {
-	rc := &RateConverter{
+	return &RateConverter{
 		httpClient:          httpClient,
-		done:                make(chan bool),
-		updateNotifier:      updateNotifier,
 		fetchingInterval:    fetchingInterval,
 		staleRatesThreshold: staleRatesThreshold,
 		syncSourceURL:       syncSourceURL,
 		rates:               atomic.Value{},
 		lastUpdated:         atomic.Value{},
 		constantRates:       NewConstantRates(),
+		time:                &timeutil.RealTime{},
 	}
+}
 
-	// In case host do not want to support currency lookup
-	// we just stop here and do nothing
-	if rc.fetchingInterval == time.Duration(0) {
-		return rc
-	}
-
-	rc.Update()                   // Make sure to populate data before to return the converter
-	go rc.startPeriodicFetching() // Start periodic ticking
-
-	return rc
+// NewRateConverterDefault returns a RateConverter with default values.
+// By default there will be no currencies conversions done.
+// `currencies.ConstantRate` will be used.
+func NewRateConverterDefault() *RateConverter {
+	return NewRateConverter(&http.Client{}, "", time.Duration(0), time.Duration(0))
 }
```

Comment:
This function seems to not be used anywhere else but in `_test.go` after your refactor. Do you think we can safely get rid of it?

Reply:
I don't think this function was ever used in production code. I deleted it and replaced the calls with calls to `NewRateConverter`.